### ISSUE TITLE
About a minute after launching, an "Application Error" dialog appears.

### ISSUE THREAD

**Issue by Lukas1h** [Lukas1h/KPPLauncher#1]
About a minute after launching, an **"Application Error"** dialog appears.

After closing the dialog, it doesn't seem to affect the app itself in any way. 

Probably a issue with [KPPLauncher](https://github.com/Lukas1h/KPPLauncher).

_Originally posted by @Lukas1h in https://github.com/Lukas1h/react-native-kindle-sampleapp/issues/1_

**Comment by Lukas1h** [comment]
Verified to be a problem with KPPLauncher.

This is right when the dialog appears.

```
[root@kindle root]# showlog -f

##Trimmed beginning##

[!]230525:103810 KPPSimpleLauncher: C :::[9733][2023-05-25 17:38:10.320] NativeThreadFreezeDetection:154 CRITICAL: Native thread ping try: 1 normal behaviour as long as pings have not exceeded 3
[!]230525:103810 KPPSimpleLauncher: C :::[9733][2023-05-25 17:38:10.320] NativeThreadFreezeDetection:123 CRITICAL: Native thread ping back, reset counter.
   230525:103811 cvm[5078]: I HeapMonitor:Information::### CC DB SIZE397312
   230525:103811 cvm[5078]: I HeapMonitor:HeapInfo:max=47448064,alloc=47448064,avail=30372136:
[!]230525:103813 KPPSimpleLauncher: C :::[9733][2023-05-25 17:38:13.243] NativeThreadFreezeDetection:168 CRITICAL: No new UI event received, stop native thread freeze detection and wait for the next event, normal execution
   230525:103841 cvm[5078]: I HeapMonitor:Information::### CC DB SIZE397312
   230525:103841 cvm[5078]: I HeapMonitor:HeapInfo:max=47448064,alloc=47448064,avail=30367960:
   230525:103841 appmgrd[2065]: I appmgr_lipc:default_alert_displayed:app_lipc_id=com.mobileread.ixtab.kpplauncher:show_simple_alert
   230525:103841 appmgrd[2065]: I lipc:evts:name=appFailedInGo, origin=com.lab126.appmgrd, fparam=com.mobileread.ixtab.kpplaunche:Event sent
   230525:103841 appmgrd[2065]: I lipc:evts:name=appActivating, origin=com.lab126.appmgrd, fparam=-1:Event sent
   230525:103841 appmgrd[2065]: W appmgr_history:timeout_check:lipc_id=com.mobileread.ixtab.kpplauncher,state=GO_SENT:Timed out waiting for app to finish GO
   230525:103841 appmgrd[2065]: I lipc:evts:name=appStateChange, origin=com.lab126.appmgrd, fparam=com.mobileread.ixtab.kpplaunche:Event sent
[!]230525:103841 [ 1486.483104] Thread-75 (5480) segfault pc: 40e14644 addr:00000000
   230525:103841 cvm[5078]: I LipcImpl:Information::Added Event : Source - com.lab126.appmgrd , Event Name - appFailedInGo : Time Start - 1685036321963, End - 1685036321963
   230525:103841 cvm[5078]: I ContentPreLoadListener:Information::Preloader weblab truned off or Child profile
   230525:103841 cvm[5078]: I LipcImpl:Information::Fired Event : Source - com.lab126.appmgrd , Event Name - appFailedInGo : Time Start - 1685036321963, End - 1685036321964
[!]230525:103841 cvm[5078]: I ChromeBarController:AppIdChange:prevApp=,curApp=:
   230525:103841 cvm[5078]: E ChromeWorkerThread:Error::Exception occured when running ChromeTask. Task = com.amazon.kindle.restricted.device.chrome.ChromeBarController$a params = com.amazon.kindle.util.lipc.LipcEvent[source=LipcTarget for:com.lab126.appmgrd associated with:com.lab126.kaf]
[!]230525:103841 cvm[5078]: java.lang.NullPointerException
[!]230525:103841 cvm[5078]: 	at com.amazon.kindle.restricted.device.chrome.ChromeBarController.a(ChromeBarController.java:234)
[!]230525:103841 cvm[5078]: 	at com.amazon.kindle.restricted.device.chrome.ChromeBarController.cUw(ChromeBarController.java:225)
[!]230525:103841 cvm[5078]: 	at com.amazon.kindle.restricted.device.chrome.ChromeBarController.cUx(ChromeBarController.java:325)
[!]230525:103841 cvm[5078]: 	at com.amazon.kindle.restricted.device.chrome.ChromeBarController$a.q(ChromeBarController.java:431)
[!]230525:103841 cvm[5078]: 	at com.amazon.kindle.restricted.device.chrome.ChromeBarController$a.as(ChromeBarController.java:418)
[!]230525:103841 cvm[5078]: 	at com.amazon.kindle.restricted.device.chrome.ChromeWorkerThread.run(ChromeWorkerThread.java:101)
[!]230525:103841 cvm[5078]: 	at java.lang.Thread.run(Unknown Source)
   230525:103841 cvm[5078]: I LipcImpl:Information::Added Event : Source - com.lab126.appmgrd , Event Name - appActivating : Time Start - 1685036321970, End - 1685036321970
   230525:103841 cvm[5078]: I ContentPreLoadListener:Information::Preloader weblab truned off or Child profile
   230525:103841 cvm[5078]: I LipcImpl:Information::Fired Event : Source - com.lab126.appmgrd , Event Name - appActivating : Time Start - 1685036321970, End - 1685036321970
   230525:103841 cvm[5078]: I LipcImpl:Information::Added Event : Source - com.lab126.appmgrd , Event Name - appStateChange : Time Start - 1685036321988, End - 1685036321988
   230525:103841 cvm[5078]: I ContentPreLoadListener:Information::Preloader weblab truned off or Child profile
   230525:103841 cvm[5078]: I LipcImpl:Information::Fired Event : Source - com.lab126.appmgrd , Event Name - appStateChange : Time Start - 1685036321988, End - 1685036321989
   230525:103841 cvm[5078]: I BookletImplementation:LifecycleEvent:stage=unload:New Lifecycle event added
[!]230525:103841 winmgr[3044]: I WindowManager:handleAppLaunchFail:unpause display and clear timeouts:
[!]230525:103841 KPPMainApp: C :::[5555][2023-05-25 17:38:41.972] ModuleManager:33 CRITICAL: Send Event: Event:0xd2e410, Owner:2, Id:1, Src:2, Dst:1, Callback:False
[!]230525:103841 KPPMainApp: C :::[5555][2023-05-25 17:38:41.972] ModuleManager:188 CRITICAL: Invoke Module:1 for Event:0xd2e410, Owner:2, Id:1, Src:2, Dst:1, Callback:False
[!]230525:103842 KPPMainApp: C :::[5555][2023-05-25 17:38:41.972] KPPChromeDataProvider:894 CRITICAL: Received app state changed event appId: com.mobileread.ixtab.kpplauncher eventType: go , value: -1, viewName:
[!]230525:103842 KPPMainApp: C :::[5555][2023-05-25 17:38:41.972] KPPChromeDataProvider:1092 CRITICAL: App failed in go, reverting app ID from com.mobileread.ixtab.kpplauncher to com.lab126.KPPMainApp and view name from  to KPP_LIBRARY
[!]230525:103842 KPPMainApp: I :::[5555][2023-05-25 17:38:41.976] KPPChromeDataProvider:807 INFO_SENSITIVE: GIsD9dn8seNWmmSiubj3GDPG0WjO6f8wB+TfMWlTPRRlFKzjYJRl9dEPBEcJTKLqL0r9nnBMbB66Behs4haEQYN6WsZ+qIFFK8ovVjChwnDKr7xcowZOjzVRTOTL91nv5U5fdCrQAkAm2++DiFtYSqlGhpn6KaHIEHHVzVDxvqpR2Q10wraWaNBF0gW6Cmbzls+Ye1uPqnaFUM8xVCTCzbnmQkaoUAeWnSCd4YGkRRGvHJG0FOAZYOIZ0gH8R0jMCc2kgF7jrJBL1NatbU2ehYygMSMAkSoytC8AwR44nmnROzk05Y6C2A0cs+huLvkCCKxBkcWQgz2czXi/uo6hu6bBtKcBAgFL2k1xw5FWe7cotm3KfKPvaPoMvk6rCWCryGnYb9iTRxmBoyAsBfc3QUSnD5unalwyNcx/NQOucog2OlfDwbsmMlyiJrhsmcY2c4hhJi848n4NHLPobi75AgisQZHFkIM9oigq8qorfcYHT/ZtfUqloEOvwjmTtpKk8l+R75OqZ4brpFBoreTmMOv0QJaxAb10qnhJnBsL/J3ai0TvAIgY+UVoiwfBzi9/+oB5yXgwnn1g8nA6jPGxCxJXDJwdBKSV2Bc9ZLjH88lozDqQPHj1iOWuIe6BE9EUeUt5MW7Rh0Kks/VH1tjjcflBgubi0rKrgvwiCQQG4o2B+aYK3tWqM6nlvsY28xIVxrPBCvXxmLB8bxt6wyIusmcLEA59Apwj2xYDIRWLxqBOPie4APUfwAsdkvleJDJ/vMbu0C9OFjlRkSgGdU88tbCphOjOAQaRkFKMHGKzKEk=
[!]230525:103842 KPPMainApp: C :::[5555][2023-05-25 17:38:42.9] ModuleManager:33 CRITICAL: Send Event: Event:0xd2e410, Owner:2, Id:1, Src:2, Dst:1, Callback:False
[!]230525:103842 KPPMainApp: C :::[5555][2023-05-25 17:38:42.10] ModuleManager:188 CRITICAL: Invoke Module:1 for Event:0xd2e410, Owner:2, Id:1, Src:2, Dst:1, Callback:False
[!]230525:103842 KPPMainApp: C :::[5555][2023-05-25 17:38:42.10] KPPChromeDataProvider:894 CRITICAL: Received app state changed event appId: com.mobileread.ixtab.kpplauncher eventType: unload , value: 0, viewName:
   230525:103842 cvm[5078]: I LipcImpl:Information::Added Event : Source - com.lab126.appmgrd , Event Name - appStateChange : Time Start - 1685036322012, End - 1685036322012
   230525:103842 cvm[5078]: I ContentPreLoadListener:Information::Preloader weblab truned off or Child profile
   230525:103842 cvm[5078]: I LipcImpl:Information::Fired Event : Source - com.lab126.appmgrd , Event Name - appStateChange : Time Start - 1685036322012, End - 1685036322013
   230525:103842 appmgrd[2065]: I appmgr_history:app_state_change:lipc_id=com.mobileread.ixtab.kpplauncher,state=UNLOAD_SENT:
   230525:103842 appmgrd[2065]: I lipc:evts:name=appStateChange, origin=com.lab126.appmgrd, fparam=com.mobileread.ixtab.kpplaunche:Event sent
   230525:103842 appmgrd[2065]: I appmgr_history:unload_app_success:app_lipc_id=com.mobileread.ixtab.kpplauncher,pid=-1:
   230525:103842 pillow[9692]: W def:pillow::jsutils_sendToJS can not send
   230525:103842 pillow[9692]: W def:pillow-bb-no-buttons::
   230525:103842 pillow[9692]: I def:show-system-alert:id=appmgrAppFailedFatal, altId=appmgrAppFailedFatal, alertLogInfo=undefined:
   230525:103842 pillow[9692]: I def:pillow::pillow window is not a window
   230525:103842 pillow[9692]: I def:pillow::pillow window is not a window
[!]230525:103842 KPPSimpleLauncher: C :::[9733][2023-05-25 17:38:42.449] NativeThreadFreezeDetection:190 CRITICAL: Starting native thread freeze detection, this is a normal preventative check
[!]230525:103842 KPPSimpleLauncher: C :::[9733][2023-05-25 17:38:42.454] NativeThreadFreezeDetection:123 CRITICAL: Native thread ping back, reset counter.
^C[root@kindle root]#

```

**Comment by Lukas1h** [comment]
Temporarily fixed with 5a0716f. Stopped the booklet from waiting for the KPP application to close before closing. Makes the opening a bit prettier, but it appears that there's a problem with long-running booklets in the background. I might try to make the booklet close after a delay instead of just immediately closing on start.
